x/evm/keeper/state_transition.go

```diff
@@ -126,19 +126,22 @@ func (k Keeper) GetHashFn() vm.GetHashFunc {
 func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumTxResponse, error) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), types.MetricKeyTransitionDB)
 
-	params := k.GetParams(k.ctx)
+	params := k.GetParams(k.Ctx())
 	ethCfg := params.ChainConfig.EthereumConfig(k.eip155ChainID)
 
 	// get the latest signer according to the chain rules from the config
-	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(k.ctx.BlockHeight()))
+	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(k.Ctx().BlockHeight()))
 
 	msg, err := tx.AsMessage(signer)
 	if err != nil {
 		return nil, stacktrace.Propagate(err, "failed to return ethereum transaction as core message")
 	}
 
 	// we use a cached context to avoid modifying to state in case EVM msg is reverted
-	commit := k.BeginCachedContext()
+	snapshot := k.Snapshot()
```

Comment:
do we need this here? the EVM already calls `Snapshot` during the state transition

Reply:
how about cost ?

Reply:
> do we need this here? the EVM already calls Snapshot during the state transition

yeah, looks like that's the case.

Reply:
removed, seems to pass all the tests.